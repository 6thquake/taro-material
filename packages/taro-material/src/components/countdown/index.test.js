import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'

import AtCountdown from '../../../.temp/components/countdown/index'

describe('AtCountdown Snap', () => {
  it('render initial AtCountdown', () => {
    const component = renderToString(<AtCountdown />)
    expect(component).toMatchSnapshot()
  })

  it('render AtCountdown -- props className', () => {
    const component = renderToString(<AtCountdown className='test' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtCountdown -- props className', () => {
    const component = renderToString(<AtCountdown customStyle='color:red;' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtCountdown -- props isCard', () => {
    const component = renderToString(<AtCountdown isCard />)
    expect(component).toMatchSnapshot()
  })

  it('render AtCountdown -- props format', () => {
    const component = renderToString(<AtCountdown format={{ hours: ':', minutes: ':', seconds: '' }} />)
    expect(component).toMatchSnapshot()
  })

  it('render AtCountdown -- props day', () => {
    const component = renderToString(<AtCountdown day={1} />)
    expect(component).toMatchSnapshot()
  })

  it('render AtCountdown -- props hours', () => {
    const component = renderToString(<AtCountdown hours={1} />)
    expect(component).toMatchSnapshot()
  })

  it('render AtCountdown -- props format', () => {
    const component = renderToString(<AtCountdown minutes={1} />)
    expect(component).toMatchSnapshot()
  })

  it('render AtCountdown -- props seconds', () => {
    const component = renderToString(<AtCountdown seconds={1} />)
    expect(component).toMatchSnapshot()
  })

  it('render AtCountdown -- props isShowDay', () => {
    const component = renderToString(<AtCountdown isShowDay />)
    expect(component).toMatchSnapshot()
  })
})
