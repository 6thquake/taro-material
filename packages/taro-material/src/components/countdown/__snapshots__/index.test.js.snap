// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtCountdown Snap render AtCountdown -- props className 1`] = `"<div class=\\"at-count-down test\\">undefined<div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">时</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">分</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">秒</span></div></div>"`;

exports[`AtCountdown Snap render AtCountdown -- props className 2`] = `"<div style=\\"color:red;\\" class=\\"at-count-down\\">undefined<div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">时</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">分</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">秒</span></div></div>"`;

exports[`AtCountdown Snap render AtCountdown -- props day 1`] = `"<div class=\\"at-count-down\\">undefined<div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">时</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">分</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">秒</span></div></div>"`;

exports[`AtCountdown Snap render AtCountdown -- props format 1`] = `"<div class=\\"at-count-down\\">undefined<div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">:</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">:</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\"></span></div></div>"`;

exports[`AtCountdown Snap render AtCountdown -- props format 2`] = `"<div class=\\"at-count-down\\">undefined<div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">时</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">01</span></div><span class=\\"taro-text at-count-down__separator\\">分</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">秒</span></div></div>"`;

exports[`AtCountdown Snap render AtCountdown -- props hours 1`] = `"<div class=\\"at-count-down\\">undefined<div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">01</span></div><span class=\\"taro-text at-count-down__separator\\">时</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">分</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">秒</span></div></div>"`;

exports[`AtCountdown Snap render AtCountdown -- props isCard 1`] = `"<div class=\\"at-count-down at-count-down--card\\">undefined<div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">时</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">分</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">秒</span></div></div>"`;

exports[`AtCountdown Snap render AtCountdown -- props isShowDay 1`] = `"<div class=\\"at-count-down\\"><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">天</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">时</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">分</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">秒</span></div></div>"`;

exports[`AtCountdown Snap render AtCountdown -- props seconds 1`] = `"<div class=\\"at-count-down\\">undefined<div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">时</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">分</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">01</span></div><span class=\\"taro-text at-count-down__separator\\">秒</span></div></div>"`;

exports[`AtCountdown Snap render initial AtCountdown 1`] = `"<div class=\\"at-count-down\\">undefined<div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">时</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">分</span></div><div class=\\"at-count-down__item\\"><div class=\\"at-count-down__time-box\\"><span class=\\"taro-text at-count-down__time\\">00</span></div><span class=\\"taro-text at-count-down__separator\\">秒</span></div></div>"`;
